### PR TITLE
Correct image stretching for helioviewer project JP2 images

### PR DESCRIPTION
**Pull request by wafels**
This solves issue #1495. This issue arises because the sunpy map object currently has no simple way of distinguishing whether the map created comes from a JP2 file or a FITS file.  The implicit assumption in map is that the data comes from a FITS file.  Image plotting therefore applies the stretching functions as described in the map source.  However, the image data in a helioviewer project JP2 file has been linearly stretched on creation.  Since map does not know the image data came from a JP2 file, the FITS stretching functions are applied and the resulting plots look terrible.  This PR solves this by adding a helioviewer Boolean valued keyword defined when reading in any JPEG2000 file passed to map, and then another function that changes the default instrument map scaling dependent on the presence/value of the keyword.

This PR is made for two reasons (1) it solves #1495 and (2) in the hope someone can come up with a better solution!

### CODE REVIEW

**Comment by wafels** [comment]
The failures are due to glymur not finding OpenJPEG.  Note that glymur now has a conda based installation - see https://glymur.readthedocs.io/en/latest/introduction.html#glymur-installation

File: sunpy/map/sources/source_type.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    Returns
+    -------
+    If the data of the map comes from the Helioviewer Project, then True is
+    returned.  If not, False is returned.
+
+    """
+    if 'helioviewer' in meta.keys():
+        return True
+    else:
+        return False
```
Comment (dpshelio) [inline sunpy/map/sources/source_type.py]: cannot just do:
```python
return 'helioviewer' in meta.keys()
```

**Comment by Cadair** [comment]
There appears to be real test failures?

**Comment by nabobalis** [comment]
Well I just reset the ones that failed.

I will need to fix those net tests this week.

**Comment by wafels** [comment]
There were real test failures.  I fixed them.

The only reported failure is in
https://travis-ci.org/sunpy/sunpy/jobs/233655080 , Python 3.4 with
NUMPY_VERSION=$PREVIOUS_NUMPY_VERSION

* File
"/home/travis/miniconda/envs/test/lib/python3.4/site-packages/glymur/version.py",
line 69, in <module>*

* kwargs['elxml'] = lxml.etree.__version__*

*NameError: name 'lxml' is not defined*

The most recent version of glymur 0.8.10 wraps this attempt to get the
version of lxml in a try/except clause.  That should be much safer.
Although... https://github.com/quintusdias/glymur/issues/409 . So maybe
there is something weird going in with Numpy 1.10.4 and/or Python 3.4 (
https://github.com/sunpy/sunpy/blob/master/.travis.yml)???

**Comment by Cadair** [comment]
arrrgghhh how is that bug back?! @quintusdias I thought we fixed this?!

**Pull request merged**